Update 0061-allow-custom-task-to-be-embedded-in-pipeline.md (#2)
ScrapCodes/community: teps/0061-allow-custom-task-to-be-embedded-in-pipeline.md

Tektoncd/Pipeline currently allows custom task to be referenced in pipeline
resource specification file using [`taskRef`](https://github.com/tektoncd/community/blob/main/teps/0002-custom-tasks.md).

> This TEP discusses the various aspects of embedding the custom task in the `RunSpec`. Just as a regular task, can be either referenced

Reviewer: ```suggestion
This TEP discusses the various aspects of embedding the custom task in the `TaskSpec` for Tekton Pipeline CRD and `RunSpec` for Tekton Run CRD. Just as a regular task, can be either referenced
```

> or embedded in the `pipelineRun`, after implementation of this TEP, a simillar support will be available for custom task as well.

Reviewer: ```suggestion
or embedded in the `pipelineRun`, after implementation of this TEP, a similar support will be available for custom task controller as well.
```

## Motivation

A custom task reference needs to be submitted to kubernetes along with
the submission of the [Tektoncd/pipeline](https://github.com/tektoncd/pipeline).

> To run the pipeline, custom task resource object creation is submitted as a separate requests to Kubernetes.

Reviewer: ```suggestion
To run the pipeline, custom task resource object creation is submitted as a separate request to Kubernetes.
```

information to render in our API/UI. 

For example, when we create a pipelineRun with custom
tasks, the KFP client first needs to make multiple API calls to Kubernetes to create all the
custom task CRDs on the same namespace before creating the `pipelineRun`. Having all the spec
inside a single `pipelineRun` can simplify task/pipeline submission for the KFP client and reduce the
number of API calls to the Kubernetes cluster. 

Currently TektonCD/Pipeline supports task specifications to be embedded in
a pipeline for regular task, but not for custom task. If Tektoncd/Pipeline
also allows a custom task specification to be embedded in a pipeline specification
then the behavior will be unified with regular task, retaining the existing the behavior of `taskRef`. 
The embedding of spec avoids the issues related to naming conflict, when multiple users in the
same namespace create resource. Related issue 
[tektoncd/pipeline#3682](https://github.com/tektoncd/pipeline/issues/3682)

### Goals

1. Allow custom tasks to be embedded in a pipeline specification.
2. Custom taskSpec should be submitted as part of the runSpec.

> 3. Document, general advice on validation/verification of custom task, to the custom task developers.

Reviewer: ```suggestion
3. Document, general advice on validation/verification of custom task, to the custom task controller developers.
```

### Non-Goals

> 1. Custom task controllers are developed by parties. Whether a custom task

Author: Note to self, re phrase this.
